src/WinXP/apps/DisplayProperties/ScreenSaverTab.js

```diff
@@ -1,32 +1,132 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import styled from 'styled-components';
 
-import display from '../../../assets/properties/displayProperties/display.png';
+import display from 'assets/properties/displayProperties/display.png';
+import arrowDown from 'assets/properties/displayProperties/icons/arrowDown.png';
+import Button from 'components/Button';
+import NumberInput from 'components/NumberInput';
+import CheckBox from 'components/CheckBox';
+import ScreenSaver from 'components/ScreenSavers';
+import { SCREEN_SAVER } from './utils';
+import { SCREEN_SAVER_PREVIEW } from 'WinXP/constants/actions';
+
+function ScreenSaverTab({ state, dispatch, appContext }) {
+  const { value, wait } = state.displayProperties.screenSaver;
+  const [screenSaverState, setScreenSaverState] = useState({ value, wait });
+  const [isNone, setIsNone] = useState(false);
+
+  useEffect(() => {
+    dispatch({ type: SCREEN_SAVER, payload: screenSaverState });
+  }, [screenSaverState, dispatch]);
+
+  useEffect(() => {
+    setIsNone(screenSaverState.value === '(None)' ? true : false);
+  }, [screenSaverState.value]);
+
+  const handleSelectChange = e => {
+    setScreenSaverState(prev => ({ ...prev, value: e.target.value }));
+  };
+
+  const handleWaitingTime = wait => {
+    setScreenSaverState(prev => ({ ...prev, wait }));
+  };
+
+  const [isKeySpacePressed, setIsKeySpacePressed] = useState(false);
+
+  //SpaceBar trigger onClick(handlePreviewOpen), which cause unwanted result.
+  // "if(isKeySpacePressed).." prevent the spacebar from triggering the onClick.
+  const handlePreviewOpen = () => {
+    if (isKeySpacePressed) setIsKeySpacePressed(false);
+    else {
+      appContext.dispatch({
+        type: SCREEN_SAVER_PREVIEW,
+        payload: value,
+      });
+    }
+  };
+
+  const handleKeyUp = e => {
+    if (e.key === ' ') setIsKeySpacePressed(true);
+  };
 
-function ScreenSaverTab() {
   return (
-    <ScreenSaver>
+    <ScreenSaverSettings>
       <div className="preview">
         <img src={display} alt="display" />
-        <div className="display-overlay"></div>
+        <div className="display-overlay">
+          <ScreenSaver
+            selectedScreenSaver={value}
+            previewScreen={true}
```

Comment:
No need for a value for boolean props, just pass it:
```suggestion
            previewScreen
```